dotcom-rendering/src/components/TopBarSupport.importable.tsx

```diff
@@ -0,0 +1,190 @@
+	HeaderPayload,
+	ModuleData,
+	ModuleDataResponse,
+} from '@guardian/support-dotcom-components/dist/dotcom/src/types';
+import { useState } from 'react';
+import { submitComponentEvent } from '../client/ophan/ophan';
+import {
+	getLastOneOffContributionDate,
+	getPurchaseInfo,
+	MODULES_VERSION,
+	shouldHideSupportMessaging,
+} from '../lib/contributions';
+import type { AuthStatus } from '../lib/identity';
+import { setAutomat } from '../lib/setAutomat';
+import { useAuthStatus } from '../lib/useAuthStatus';
+import { useCountryCode } from '../lib/useCountryCode';
+import { useOnce } from '../lib/useOnce';
+import { usePageViewId } from '../lib/usePageViewId';
+import { useConfig } from './ConfigContext';
+
+type Props = {
+	contributionsServiceUrl: string;
+};
+
+const headerStyles = css`
+	padding-top: 39px;
+	padding-left: 10px;
+	max-width: 310px;
+
+	${from.mobileMedium} {
+		padding-top: 44px;
+	}
+
+	${from.mobileLandscape} {
+		padding-left: ${space[5]}px;
+	}
+
+	${from.tablet} {
+		padding-top: ${space[1]}px;
+		padding-bottom: ${space[3]}px;
+		padding-left: ${space[5]}px;
+		max-width: 400px;
+	}
+
+	${from.desktop} {
+		max-width: none;
+	}
+`;
+
+function getIsSignedIn(authStatus: AuthStatus): boolean | undefined {
+	switch (authStatus.kind) {
+		case 'Pending':
+			return undefined;
+		case 'SignedInWithCookies':
+		case 'SignedInWithOkta':
+			return true;
+		case 'SignedOutWithCookies':
+		case 'SignedOutWithOkta':
+			return false;
+	}
+}
+
+type ReaderRevenueLinksRemoteProps = {
+	countryCode: string;
+	pageViewId: string;
+	contributionsServiceUrl: string;
+};
+
+const ReaderRevenueLinksRemote = ({
+	countryCode,
+	pageViewId,
+	contributionsServiceUrl,
+}: ReaderRevenueLinksRemoteProps) => {
+	const [supportHeaderResponse, setSupportHeaderResponse] =
+		useState<ModuleData | null>(null);
+	const [SupportHeader, setSupportHeader] =
+		useState<React.ElementType | null>(null);
+	const isSignedIn = getIsSignedIn(useAuthStatus());
+
+	const { renderingTarget } = useConfig();
+
+	useOnce((): void => {
+		setAutomat();
+
+		const requestData: HeaderPayload = {
+			tracking: {
+				ophanPageId: pageViewId,
+				platformId: 'GUARDIAN_WEB',
+				referrerUrl: window.location.origin + window.location.pathname,
+				clientName: 'dcr',
+			},
+			targeting: {
+				showSupportMessaging: !shouldHideSupportMessaging(),
+				countryCode,
+				modulesVersion: MODULES_VERSION,
+				mvtId: Number(
+					getCookie({ name: 'GU_mvt_id', shouldMemoize: true }),
+				),
+				lastOneOffContributionDate: getLastOneOffContributionDate(),
+				purchaseInfo: getPurchaseInfo(),
+				isSignedIn: isSignedIn === true,
+			},
+		};
+		getHeader(contributionsServiceUrl, requestData)
+			.then((response: ModuleDataResponse) => {
+				if (!response.data) {
+					return null;
+				}
+
+				const { module } = response.data;
+				setSupportHeaderResponse(module);
+
+				return import(
+					/* webpackChunkName: "header" */ `./marketing/header/Header`
```

Comment:
🎉

Reply:
ah yes, I'd forgotten that was separate.
Perhaps we can use [`module.name`](https://github.com/guardian/support-dotcom-components/blob/main/packages/server/src/api/headerRouter.ts#L64) in the response to pick the right component?